### DISCUSSION TITLE
Got maximum execution time error even there is no job to do

### DISCUSSION THREAD

**Discussion by seyyedam7** [laravel/framework#39961] [Q&A]
- Laravel Version: 8.75.0
- PHP Version: 7.4.12
- Database Driver & Version: Redis v3.2.1

### Description:
I ran queue listener with these options:
`php artisan queue:work --queue=default --timeout=0 --sleep=5`

Command works fine when we haven't anything to do.
But when all the jobs were done, I faced with these errors after about 2 minutes.
I think there are some issues with the sleep method in the queue listener.

![image](https://user-images.githubusercontent.com/51041077/145428612-2a036ece-370c-40f9-a659-6439bf4b3a70.png)
![image](https://user-images.githubusercontent.com/51041077/145428689-48cfec0c-30d3-40e8-8631-a7639b208b29.png)

**Comment by erikn69** [comment]
Maximum execution time on `php.ini`

**Reply by seyyedam7** [reply]
Yeah, I know that, but this is not the problem.
I used to do my jobs with this command:
`php artisan queue:listen`
But in last update, this command does only one job and I don't know why this happens.
So I decided to use
`php artisan queue:work`
and it works fine.
But when all jobs done and there are no jobs to do in database, I faced with these errors.
I think there is some issue with sleep method in queue:work.
Because it works fine when I start this command and I can wait indefinitely until jobs table is empty.
When I define a job, it will be done and completed successfully.
But sleep method doesn't work anymore and queue:work can't wait for the next time (it depends on max_executation_time to get errors)
